src/main/java/com/kryptnostic/rhizome/hazelcast/processors/AbstractRhizomeEntryProcessor.java

```diff
@@ -20,24 +19,26 @@ public AbstractRhizomeEntryProcessor() {
         this( true );
     }
 
-    @Override
-    public abstract R process( @Nonnull Map.Entry<K, V> entry );
-
     public AbstractRhizomeEntryProcessor( boolean applyOnBackup ) {
         this.applyOnBackup = applyOnBackup;
     }
 
     @Override
-    public final EntryBackupProcessor<K, V> getBackupProcessor() {
+    public abstract R process( @Nonnull Map.Entry<K, V> entry );
+
+    @Override
+    public final EntryProcessor<K, V, R> getBackupProcessor() {
         if ( applyOnBackup ) {
             return this;
         }
         return null;
     }
 
-    @Override
-    public void processBackup( Entry<K, V> entry ) {
-        process( entry );
+    @Override public R process( MutableEntry<K, V> entry, Object... arguments ) throws EntryProcessorException {
+        return process( entry );
     }
 
+    @Override public javax.cache.processor.EntryProcessor<K, V, R> createBackupEntryProcessor() {
```

Comment:
um. what is this?

Reply:
It took me a bit to figure it out as well. Hazelcast has decided to have backup entry processors implement JSR-107/SPI 1.0.0 for some reason. So previously we did something similar by dispatching entry processor to primary and backup entries  (in comparison to apply entry processor to local entry and then shipping updated value to back up). This is preferred, especially for large entities with small transformations since shipping a big entity to capture a tiny change is bad. 

Hazelcast implemented this a javax.cache.processor that gets serialized to backup entry (but since EP extends processor you can return this).